bin/find-spark-home

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Attempts to find a proper value for SPARK_HOME
+
+FIND_SPARK_HOME_PYTHON_SCRIPT="$(cd "`dirname "$0"`"; pwd)/find_spark_home.py"
+
+if [ ! -z "${SPARK_HOME}" ]; then
+   exit 0
+elif [ ! -f $FIND_SPARK_HOME_PYTHON_SCRIPT ]; then
+  # If we are not in the same directory as find_spark_home.py we are not pip installed
+  export SPARK_HOME="$(cd "`dirname "$0"`"/..; pwd)"
+else
+  # We are pip installed, use the Python script to resolve a reasonable SPARK_HOME
+  # Default to standard python interpreter unless told otherwise
+  if [[ -z "$PYSPARK_DRIVER_PYTHON" ]]; then
```

Comment:
It does - so the reasoning behind this is that this is called before the Python runner and if the user has a specific driver Python they want to use we should resolve PySpark's location using that (e.g. if the user has both Python 2 and Python 3 envs setup). We could probably 86 this and trust the user to have the preffered Python first in path (e.g. by activating the correct virtualenv) but I figured it was safer to respect PYSPARK_DRIVER_PYTHON if set than not.